server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/FixedLengthStarTreeNode.java

```diff
@@ -0,0 +1,186 @@
+    private static final int END_DOC_ID_OFFSET = START_DOC_ID_OFFSET + Integer.BYTES;
+    private static final int AGGREGATE_DOC_ID_OFFSET = END_DOC_ID_OFFSET + Integer.BYTES;
+    private static final int IS_STAR_NODE_OFFSET = AGGREGATE_DOC_ID_OFFSET + Byte.BYTES;
+    private static final int FIRST_CHILD_ID_OFFSET = IS_STAR_NODE_OFFSET + Integer.BYTES;
+    private static final int LAST_CHILD_ID_OFFSET = FIRST_CHILD_ID_OFFSET + Integer.BYTES;
+
+    public static final int INVALID_ID = -1;
+
+    private final int nodeId;
+    private final int firstChildId;
+
+    RandomAccessInput in;
+
+    public FixedLengthStarTreeNode(RandomAccessInput in, int nodeId) throws IOException {
+        this.in = in;
+        this.nodeId = nodeId;
+        firstChildId = getInt(FIRST_CHILD_ID_OFFSET);
+    }
+
+    private int getInt(int fieldOffset) throws IOException {
+        return in.readInt(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    private long getLong(int fieldOffset) throws IOException {
+        return in.readLong(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    private byte getByte(int fieldOffset) throws IOException {
+        return in.readByte(nodeId * SERIALIZABLE_DATA_SIZE_IN_BYTES + fieldOffset);
+    }
+
+    @Override
+    public int getDimensionId() throws IOException {
+        return getInt(DIMENSION_ID_OFFSET);
+    }
+
+    @Override
+    public long getDimensionValue() throws IOException {
+        return getLong(DIMENSION_VALUE_OFFSET);
+    }
+
+    @Override
+    public int getChildDimensionId() throws IOException {
+        if (firstChildId == INVALID_ID) {
+            return INVALID_ID;
+        } else {
+            return in.readInt(firstChildId * SERIALIZABLE_DATA_SIZE_IN_BYTES);
+        }
+    }
+
+    @Override
+    public int getStartDocId() throws IOException {
+        return getInt(START_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getEndDocId() throws IOException {
+        return getInt(END_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getAggregatedDocId() throws IOException {
+        return getInt(AGGREGATE_DOC_ID_OFFSET);
+    }
+
+    @Override
+    public int getNumChildren() throws IOException {
+        if (firstChildId == INVALID_ID) {
+            return 0;
+        } else {
+            return getInt(LAST_CHILD_ID_OFFSET) - firstChildId + 1;
+        }
+    }
+
+    @Override
+    public boolean isLeaf() {
+        return firstChildId == INVALID_ID;
+    }
+
+    @Override
+    public boolean isStarNode() throws IOException {
+        return getByte(IS_STAR_NODE_OFFSET) != 0;
+    }
+
+    @Override
+    public StarTreeNode getChildForDimensionValue(long dimensionValue) throws IOException {
+        // there will be no children for leaf nodes
+        if (isLeaf()) {
+            return null;
+        }
+
+        // Specialize star node for performance
+        if (dimensionValue == ALL) {
```

Comment:
Major : 
How are we ensuring star node is the first child ? 

What if there is a node with value -2.

Reply:
As mentioned above in comments, using a isStarNode flag to check whether we want to read star node or not.